src/main/java/hu/oe/nik/szfmv/automatedcar/model/WorldObject.java

```diff
@@ -231,15 +233,16 @@ public void afterUnmarshal(Unmarshaller u, Object parent) {
      * {@inheritDoc}
      */
     @Override
-    public Shape getPolygon(int offsetX, int offsetY) {
-        return this.getShapeTransfrom(offsetX, offsetY).createTransformedShape(this.polygon);
+    public List<Shape> getPolygons() {
+        return this.polygons;
+        //return this.getShapeTransfrom().createTransformedShape(this.polygon);
     }
 
-    private AffineTransform getShapeTransfrom(int offsetX, int offsetY) {
+    private AffineTransform getShapeTransfrom() {
```

Comment:
Legyen public és benne az IObjectben is. Majd setRotation setPosX és setPosY-al forgassák és mozgassák a többiek!!!

Reply:
A rajzolásnál mindig új AffineTransfrom kell az offset miatt, így a setPosX/Y stb... beállítása nem elég. Vagy járható út lenne, hogyha a kirajzolásnál csak offsettel mozgatnánk el és a többi mozgatást a worldObject megcsinálja, viszont a lista miatt problémás az elemek ottani módosítgatása.

Reply:
Lehet módosítani! t.translate(t.getTranslateX() + offsetX, t.getTranslateY() + offsetY)

Reply:
Nem értem a probléma forrását. Ezek a shapek egyben mozognak nem? Ha külön mozognak akkor pedig lehet ez is AffineTransform lista.